### PR TITLE
feat(condo): DOMA-7772 added marketplace pushes

### PR DESCRIPTION
**Pull request by VKislov**
Push to publish an invoice with and without a ticket

### CODE REVIEW

File: apps/condo/domains/marketplace/schema/Invoice.js [hunk 1]
```diff
@@ -33,10 +36,11 @@ const {
     ERROR_FORBID_EDIT_PUBLISHED,
 } = require('@condo/domains/marketplace/constants')
 const { INVOICE_ROWS_FIELD } = require('@condo/domains/marketplace/schema/fields/invoiceRows')
+const { MARKETPLACE_INVOICE_PUBLISHED_MESSAGE_TYPE, MARKETPLACE_INVOICE_WITH_TICKET_PUBLISHED_MESSAGE_TYPE } = require('@condo/domains/notification/constants/constants')
```
[the remark this answers is not in the document]
Comment (VKislov) [reply apps/condo/domains/marketplace/schema/Invoice.js:39]: This file is older than us, I didn’t come up with it)

image
image

File: apps/condo/domains/marketplace/schema/Invoice.js [hunk 2]
```diff
@@ -318,6 +322,67 @@ const Invoice = new GQLListSchema('Invoice', {
 
             return resolvedData
         },
+        afterChange: async ({ context, operation, originalInput, updatedItem }) => {
+            const userId = updatedItem.client
+            const propertyId = updatedItem.property
+            const unitName = updatedItem.unitName
+            const unitType = updatedItem.unitType
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/schema/Invoice.js:329]: You can easy single line this
```suggestion
            const { client: userId, property: propertyId, unitName, unitType } = updatedItem
```

File: apps/condo/domains/marketplace/schema/Invoice.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alllex202) [reply apps/condo/domains/marketplace/schema/Invoice.js:39]: Someday we need to find "a minute" and fix this dinosaur)

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/marketplace/schema/Invoice.test.js [hunk 3]
```diff
@@ -591,6 +595,210 @@ describe('Invoice', () => {
...
+
+            const messages = await Message.getAll(adminClient, {
+                user: { id: residentClient.user.id },
+                type: MARKETPLACE_INVOICE_PUBLISHED_MESSAGE_TYPE,
+            })
+
+            expect(messages).toHaveLength(1)
+            expect(messages[0].meta.data.invoiceId).toEqual(invoice.id)
+            expect(messages[0].meta.data.residentId).toEqual(resident.id)
+            expect(messages[0].meta.data.url).toEqual(`${conf.SERVER_URL}/invoice/${invoice.id}`)
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/schema/Invoice.test.js:632]: Maybe wrap this block with `waitFor`? If I'm correct, message API executes in worker, so you need to ensure, that task will ends with expected result

File: apps/condo/domains/marketplace/schema/Invoice.test.js [hunk 4]
```diff
@@ -591,6 +595,210 @@ describe('Invoice', () => {
...
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/schema/Invoice.test.js:674]: Same as above comment

File: apps/condo/domains/marketplace/schema/Invoice.test.js [hunk 5]
```diff
@@ -591,6 +595,210 @@ describe('Invoice', () => {
...
+                status: INVOICE_STATUS_PAID,
+                client: { connect: { id: residentClient.user.id } },
+            })
+
+            const messages = await Message.getAll(adminClient, {
+                user: { id: residentClient.user.id },
+                type: MARKETPLACE_INVOICE_PUBLISHED_MESSAGE_TYPE,
+            })
+
+            expect(messages).toHaveLength(0)
```
Comment (sitozzz) [inline apps/condo/domains/marketplace/schema/Invoice.test.js:707]: At least all of the `Message` domain related logic should be tested with delayed helper function `waitFor`

**Review by sitozzz — APPROVED** [review]

**Pull request merged**